### DISCUSSION TITLE
Is it possible to create a Custom Validator to use it in all Models?

### DISCUSSION THREAD

**Discussion by lucassimines** [sequelize/sequelize#14346] [Help & Questions] [answered]
Is it possible to create a Custom Validator to use it in all Models, like extending all default validations?

**Comment by fzn0x** [comment, marked as answer]
Yes, you can use `validate` property on the field object to do so.

Take a look: https://docs.sequelizejs.com/docs/v7/core-concepts/validations-and-constraints/#per-attribute-validations

**Reply by lucassimines** [reply]
> Yes, you can use `validate` property on the field object to do so.
> 
> Take a look: https://docs.sequelizejs.com/docs/v7/core-concepts/validations-and-constraints/#per-attribute-validations

But, how can I use my custom validator function to all models? This way I have to write the function on each model.

**Reply by fzn0x** [reply]
> > Yes, you can use the `validate` property on the field object to do so.
> > Take a look: https://docs.sequelizejs.com/docs/v7/core-concepts/validations-and-constraints/#per-attribute-validations
> 
> But, how can I use my custom validator function for all models? This way I have to write the function on each model.

You can't, you must define it for each model for validation to avoid side effects on the model you don't want to validate, so.. you can do object destructuring with javascript, you can create one file which returns global validation and just assign it to validate object without rewriting how the validation should work. (DRY solution).

**Reply by fzn0x** [reply]
e.g

```js
import globalValidation from "./globalValidation.js" // import your global validation

validate: {
 ...globalValidation, // destructure it
 // specific-model validation only is below here
}

**Reply by lucassimines** [reply]
Thanks @fzn0x for the help!
